runtime/driver/src/main/java/org/apache/nemo/driver/NemoDriver.java

```diff
@@ -142,7 +142,10 @@ private void setUpLogger() {
    */
   private void shutdown() {
     LOG.info("Driver shutdown initiated");
-    runnerThread.execute(runtimeMaster::terminate);
+    runnerThread.execute(() -> {
+      runtimeMaster.saveMetrics();
```

Comment:
Would there be a particular reason why this is not on L203?

Reply:
i try to minimize side-effects. If it looks like there is no problem to move that line, I will merge two functions, flushMetric and saveMetric

Reply:
I tried to minimize the side effect, but it seems to have been my misunderstanding. I will merge flushMetric function and saveMetric function